src/NuGet.Core/NuGet.ProjectModel/PackageSpecWriter.cs

```diff
@@ -453,6 +453,45 @@ internal static void SetDependencies(IObjectWriter writer, string name, IEnumera
             writer.WriteObjectEnd();
         }
 
+        /// <summary>
+        /// The central transitive dependecy groups are used for pack operation.
+        /// The metadata needed for pack is composed from:
+        ///     Name, IncludeType, SuppressParent and Version 
+        /// </summary>
+        internal static void SetCentralTransitveDependencyGroup(IObjectWriter writer, string name, IEnumerable<LibraryDependency> libraryDependencies)
+        {
+            if (!libraryDependencies.Any())
+            {
+                return;
+            }
+
+            writer.WriteObjectStart(name);
+
+            foreach (var dependency in libraryDependencies)
```

Comment:
I think these need to be sorted by name, otherwise no-op will likely break. 

Check what SetDependencies does.

Reply:
It does not have effect on the no-op as it is not in dgspec. Adding ordering to be reader friendlier.

Reply:
Yes, you are correct. That's twice, the same misconception on the same PR :'(. 

I recently added sorting in the equality check which decies whether we write the assets file. https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.ProjectModel/CentralTransitiveDependencyGroup.cs